File: unical-support-2/src/main/java/unical_support/unicalsupport2/data/dto/template/TemplateDto.java
package unical_support.unicalsupport2.data.dto.template;

import lombok.Data;
import unical_support.unicalsupport2.data.dto.CategoryDto;

import java.util.List;

@Data
public class TemplateDto {
    private String id;
    private String name;
    private CategoryDto category;
    private String content;
    private List<ParameterDto> parameters;
}

File: unical-support-interface/src/app/model/TemplateDto.ts
import {ParameterDto} from './ParameterDto';
import {CategoryDto} from './category-dto';

export class TemplateDto {
  id: string;
  name: string;
  content: string;
  description: string;
  category: CategoryDto;
  parameters: ParameterDto[];

  constructor(data: any) {
    this.id = data.id;
    this.name = data.name;
    this.content = data.content;
    this.description = data.description;
    this.category = data.category;
    this.parameters = data.parameters;
  }
}
